### ISSUE TITLE
Warn before running executables

### ISSUE THREAD

**Issue by drequivalent** [elementary/files#745]
It pertrains not only to Appimages, but also to other binaries that the user may run.
When clicking or hitting Enter on a binary, hash whatever file the user is trying to run against a database, and if it checks out, let it run.
If not, show the message akin to what Terminal is showing when trying to copy the sudo command:

![binary warning svg](https://user-images.githubusercontent.com/1463277/48305252-63cb5780-e538-11e8-949b-ba8997cdfa4d.png)

On "Run and remember" the binary's hash is saved into the database for future use.

The user should be able to disable it in Privacy and security Switchboard tab (or disable it for a directory) - because development requires running different binaries all the time.

**Comment by jeremypw** [comment]
@danirabbit Should we continue to support running executables from Files? It seems more risky than its worth tbh.  At present you can run a GUI executable from e.g. `/bin` OK but if it is a terminal app or script nothing appears to happen.

**Comment by danirabbit** [comment]
@jeremypw I personally would be okay with not having this. Folks can always run random executables from terminal

**Comment by jeremypw** [comment]
Yes, I am in two minds. I don't want to complicate Files further by maintaining a list of "safe to run" executables but at least if a random executable (i.e. one not in the Applications Menu) is run from Files there is an opportunity to warn compare to entering Terminal commands.
